### ISSUE TITLE
[css-anchor-position] Was changing acceptable element algorithm intentional?

### ISSUE THREAD

**Issue by anttijk** [w3c/csswg-drafts#11170]
https://github.com/w3c/csswg-drafts/commit/fe8e40d37faae "Rewrite the ordering conditions on 'acceptable anchor element' to be more explicit and hopefully clearer" actually changed the algorithm somewhat. Specifically in the "different containing blocks" case the tree order of the elements no longer matters. This affects for example

http://wpt.live/css/css-anchor-position/anchor-name-002.html (the first target now gets anchored)
http://wpt.live/css/css-anchor-position/anchor-scroll-002.html (all targets attach to the last anchor)

The commit message indicated that the intention was to clarify the algorithm so I'm wondering if the behavior changes were intentional?

**Comment by tabatkins** [comment]
Nope, losing the condition about the painting order of `possible anchor`'s penultimate CB before the shared CB was an accident.
